cowamm/profitability/competitor_kpis/curve/curve_kpis_4232873.sql

```diff
@@ -0,0 +1,31 @@
+-- Computes volume, tvl and APR for Curve pools
+-- APR is measured as the fees earned per $ invested, over the last 24 hours, projected over 1 year
+-- Input: blockchain
+select
+    r.contract_address,
+    sum(amount_usd) as volume,
+    365 * sum(amount_usd * fee / tvl) as apr,
+    avg(fee) as fee,
+    avg(tvl) as tvl
+from "query_4232976(blockchain='{{blockchain}}')" as r
+left join curve.trades as t
+    on
+        r.contract_address = t.project_contract_address
+        and r.tx_hash = t.tx_hash
+where
+    t.block_time >= date_add('day', -1, now())
+    -- This test avoids any possible issue with reconstructing the reserves of the pool
+    and tvl > 0
+group by r.contract_address
+
+union distinct
+select
+    contract_address,
+    0 as volume,
+    0 as apr,
+    fee,
+    tvl
+from "query_4232976(blockchain='{{blockchain}}')"
+where
+    time < date_add('day', -1, now())
+    and latest = 1
```

Comment:
What does this do? Is this for the case that tvl is 0? I think it would be easier to use an if statement in the apr computation line (or omit these pools altogether as they seem irrelevant).

Reply:
This for the case when the pool has not traed in the period specified by the query. It could be interesting to show that their APR is 0% to promote our pools instead.
I thought it was better to implement it in the query itself, than handling this edge case elsewhere.

Reply:
But don't you get those with the upper query as well (since it's a left join)? The only issue I see is that you may have a "divide by 0" issue in `sum(amount_usd * fee / tvl)` but there you could use something like [`try`](https://prestodb.io/docs/current/functions/conditional.html#id4) of if statements instead of needed an entire new union query.

Reply:
Yes, indeed the left join does the work. I removed this part.